sky/authentication.py

```diff
@@ -62,6 +63,7 @@
 # TODO(zhwu): Support user specified key pair.
 PRIVATE_SSH_KEY_PATH = '~/.ssh/sky-key'
 PUBLIC_SSH_KEY_PATH = '~/.ssh/sky-key.pub'
+_SSH_KEY_GENERATION_LOCK = '~/.ssh/.sky-key.lock'
```

Comment:
```suggestion

# Lock protecting public/private sky-key generation.
_SSH_KEY_GENERATION_LOCK = '~/.ssh/.sky-key.lock'
```

Reply:
How about putting all our generated files in ~/.sky/generated or a lock dir under ~/.sky?

Reply:
Good point! Moved it to `~/.sky/generated/ssh` and test it in a new environment by running 5 concurrent launch.

`for i in `seq 1 5`; do sky launch -c test-gcp-$i -d -y --cloud gcp --cpus 2 echo hi & done`